### PR TITLE
feat(clearinghouse): add auto retrigger for END_CLEARINGHOUSE

### PR DESCRIPTION
**Pull request by Phil91**
## Description

enable maintenance worker to retrigger the clearinghouse process

## Why

To enable an auto retrigger for the clearinghouse process if the END_CLEARINGHOUSE is on TODO for a certain time.

## Issue

Refs: #810

## Corresponding CD PR

https://github.com/eclipse-tractusx/portal/pull/418

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/maintenance/Maintenance.App/DependencyInjection/MaintenanceServiceExtensions.cs
Comment (evegufy) [inline src/maintenance/Maintenance.App/DependencyInjection/MaintenanceServiceExtensions.cs:1]: please add file header

File: src/maintenance/Maintenance.App/DependencyInjection/BatchDeleteServiceSettings.cs
Comment (evegufy) [inline src/maintenance/Maintenance.App/DependencyInjection/BatchDeleteServiceSettings.cs:1]: please add file header

File: src/administration/Administration.Service/Extensions/CompanyApplicationStatusFilterExtensions.cs
Comment (evegufy) [inline src/administration/Administration.Service/Extensions/CompanyApplicationStatusFilterExtensions.cs:1]: Please add file header

**Review by evegufy — CHANGES_REQUESTED** [review]
Please take care of codeql findings

File: src/administration/Administration.Service/Extensions/CompanyApplicationStatusFilterExtensions.cs
Comment (Phil91) [reply src/administration/Administration.Service/Extensions/CompanyApplicationStatusFilterExtensions.cs:1]: added

File: src/maintenance/Maintenance.App/DependencyInjection/BatchDeleteServiceSettings.cs
Comment (Phil91) [reply src/maintenance/Maintenance.App/DependencyInjection/BatchDeleteServiceSettings.cs:1]: added

File: src/maintenance/Maintenance.App/DependencyInjection/MaintenanceServiceExtensions.cs
Comment (Phil91) [reply src/maintenance/Maintenance.App/DependencyInjection/MaintenanceServiceExtensions.cs:1]: added

File: src/administration/Administration.Service/Extensions/CompanyApplicationStatusFilterExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ ********************************************************************************/
+
+using Org.Eclipse.TractusX.Portal.Backend.Administration.Service.Models;
+using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Enums;
+
+namespace Org.Eclipse.TractusX.Portal.Backend.Administration.Service.Extensions;
+
+public static class CompanyApplicationStatusFilterExtensions
+{
+    public static IEnumerable<CompanyApplicationStatusId> GetCompanyApplicationStatusIds(this CompanyApplicationStatusFilter? companyApplicationStatusFilter) =>
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/administration/Administration.Service/Extensions/CompanyApplicationStatusFilterExtensions.cs:27]: Even if it doesn't belong to this ticket - as I like this extension better then the private method I've refactored RegistrationBusinessLogic accordingly.

**Review by ntruchsess — CHANGES_REQUESTED** [review]
I've refactored some hard to read code, removed some reduncancy and moved code to where it belongs (calls on dbcontext into repo).
remark: not all scanner findings give good guidance to create better understandable code - e.g. implicitly selecting (cause of async lambda) into into a task in the foreach-loop where that task is awaited at a different location might not be understud by every developer at the first glance. I did add a new factory-method that turns IEnumerable<Task<X>> into IAsyncEnumerable<X> which allows to await the tasks directly in the foreach.
There is one comment you should check (new unreferenced setting) and 4 scanner-findings I leave up to you to resolve ;-)

**Comment by ntruchsess** [comment]
@Phil91 : had to replace END_CLEARINGHOUSE by AWAIT_CLEARING_HOUSE_RESPONSE as per rebase on main. From my understanding that should just have been a rename without semantic changes?

File: src/administration/Administration.Service/DependencyInjection/PartnerRegistrationSettings.cs [hunk 2]
```diff
@@ -28,6 +28,8 @@ public class PartnerRegistrationSettings
     [Required]
     [DistinctValues("x => x.ClientId")]
     public IEnumerable<UserRoleConfig> InitialRoles { get; set; } = null!;
+
+    public int ApplicationsMaxPageSize { get; set; }
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/administration/Administration.Service/DependencyInjection/PartnerRegistrationSettings.cs]: seems like I've added that by mistake, removed it.

**Comment by Phil91** [comment]
> @Phil91 : had to replace END_CLEARINGHOUSE by AWAIT_CLEARING_HOUSE_RESPONSE as per rebase on main. From my understanding that should just have been a rename without semantic changes?

@ntruchsess correct :)

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**